### PR TITLE
Show all search result in a leaderboard (show all possible matches)

### LINKED ISSUE #1775: Show all search results
In the Leaderboards tab, the search is done as the user types something on the keyboard, showing the first matching item and the rest of users beneath it. This is however not practical and not efficient, thus it is aimed to narrow the search results down to users matching the input pattern: 
ex. if "kr" is given, the results should look something like "krristi"; "kraboom" and so on.

### PR DESCRIPTION
**Pull request by Marc-Spector**
Closes #1775 

UI elements
image

### CODE REVIEW

**Comment by Marc-Spector** [comment]
The PR is ready to review

File: src/main/resources/theme/leaderboard/leaderboards.fxml [hunk 1]
```diff
@@ -16,7 +16,7 @@
       fx:controller="com.faforever.client.leaderboard.LeaderboardsController">
     <Separator minHeight="10.0"/>
     <HBox alignment="CENTER_LEFT" spacing="10.0">
-        <ComboBox fx:id="leaderboardComboBox" onAction="#onLeaderboardSelected"/>
```
Comment (Sheikah45) [inline src/main/resources/theme/leaderboard/leaderboards.fxml]: Why the change from the actionEvent to the listener? I would think the actionEvent would be better and less prone to leaks

File: src/main/java/com/faforever/client/leaderboard/LeaderboardsController.java [hunk 2]
```diff
@@ -125,17 +132,26 @@ public Leaderboard fromString(String string) {
...
+      searchTextField.clear();
+      if (usernamesAutoCompletion != null) {
+        usernamesAutoCompletion.dispose();
+      }
+      leaderboardService.getEntries(newValue).thenAccept(leaderboardEntryBeans -> {
+        ratingTable.setItems(observableList(leaderboardEntryBeans));
+        usernamesAutoCompletion = TextFields.bindAutoCompletion(searchTextField,
+            leaderboardEntryBeans.stream().map(LeaderboardEntry::getUsername).collect(Collectors.toList()));
+        usernamesAutoCompletion.setDelay(0);
+        contentPane.setVisible(true);
```
Comment (Sheikah45) [inline src/main/java/com/faforever/client/leaderboard/LeaderboardsController.java]: I don't see the need for the change to the listener over the onAction method

**Review by Sheikah45 — CHANGES_REQUESTED** [review]

File: src/main/java/com/faforever/client/leaderboard/LeaderboardsController.java [hunk 3]
```diff
@@ -125,17 +132,26 @@ public Leaderboard fromString(String string) {
...
-    });
+      searchTextField.clear();
+      if (usernamesAutoCompletion != null) {
+        usernamesAutoCompletion.dispose();
+      }
+      leaderboardService.getEntries(newValue).thenAccept(leaderboardEntryBeans -> {
+        ratingTable.setItems(observableList(leaderboardEntryBeans));
+        usernamesAutoCompletion = TextFields.bindAutoCompletion(searchTextField,
+            leaderboardEntryBeans.stream().map(LeaderboardEntry::getUsername).collect(Collectors.toList()));
+        usernamesAutoCompletion.setDelay(0);
```
Comment (1-alex98) [inline src/main/java/com/faforever/client/leaderboard/LeaderboardsController.java]: Don't you think this is a bad thing that the client asks the api with a request on the user table (big table) every letter you type in. Say the username is TrustTheFall. Makes 12 http requests and sql queries XD

Comment (1-alex98) [reply src/main/java/com/faforever/client/leaderboard/LeaderboardsController.java]: I mean there should definitely be a big delay to ensure the server does not get too many requests

**Review by 1-alex98 — CHANGES_REQUESTED** [review]

File: src/main/java/com/faforever/client/leaderboard/LeaderboardsController.java (hunk 3 above)
Comment (Sheikah45) [reply src/main/java/com/faforever/client/leaderboard/LeaderboardsController.java]: This action is on the comboBox choice not the textField input I believe

Comment (Sheikah45) [reply src/main/java/com/faforever/client/leaderboard/LeaderboardsController.java]: So the api is only called when the user changes the leaderboard they are searching. Then that list is used to populate the autocompletion.

Comment (1-alex98) [reply src/main/java/com/faforever/client/leaderboard/LeaderboardsController.java]: Then this only work with first 10 000 again ... well ok

Comment (1-alex98) [reply src/main/java/com/faforever/client/leaderboard/LeaderboardsController.java]: Please call oldValue and newValue with proper names then I would probably better understand what you are doing. newValue-> newlySelectedLeaderBoard

File: src/main/resources/theme/leaderboard/leaderboards.fxml (hunk 1 above)
Comment (Marc-Spector) [reply src/main/resources/theme/leaderboard/leaderboards.fxml]: To simplify the code. We don't need to manually call the `leaderboardComboBox.getValue()` method. The listener will automatically give us the newly selected value. But if this is done to be less prone to leaks, then okay.

File: src/main/java/com/faforever/client/leaderboard/LeaderboardsController.java (hunk 2 above)
Comment (Marc-Spector) [reply src/main/java/com/faforever/client/leaderboard/LeaderboardsController.java]: Okay

**Review by Sheikah45 — APPROVED** [review]
